packages/qwik/src/core/tests/inline-component.spec.tsx

```diff
@@ -419,4 +421,43 @@ describe.each([
       </Component>
     );
   });
+
+  it('should render component$ inside inlined wrapper', async () => {
+    interface ComplexWrapperProps {
+      foo: string;
+    }
+
+    const ComplexWrapper = (
+      props: PublicProps<ComplexWrapperProps>,
+      key: string | null,
+      flags: number
+    ) => {
+      const cmpFn = component$<ComplexWrapperProps>(({ foo }) => {
+        return (
+          <div>
+            {foo}: <Slot />
+          </div>
+        );
+      });
+      return cmpFn(props, key, flags);
+    };
+
+    const { vNode } = await render(
+      <ComplexWrapper foo="bar">
+        <div>
+          bar: <div id="1">Test</div>
+        </div>
```

Comment:
oh this shouldn't be here :open_mouth: that's why this test is passing

with this rendered content, the expected `vNode` should be:
```tsx
      <InlineComponent>
        <Component>
          <div>
            bar: 
            <div>
              bar: <div id="1">Test</div>
            </div>
          </div>
        </Component>
      </InlineComponent>
```
and not the one below.

Here, try with this test instead:
```tsx
  it('should render component$ inside inlined wrapper', async () => {
    interface ComplexWrapperProps {
      foo: string;
    }

    const ComplexWrapper = (
      props: PublicProps<ComplexWrapperProps>,
      key: string | null,
      flags: number
    ) => {
      const cmpFn = component$<ComplexWrapperProps>(({ foo }) => {
        return (
          <div>
            {foo}: <Slot />
          </div>
        );
      });
      return cmpFn(props, key, flags);
    };

    const { vNode } = await render(<ComplexWrapper foo="bar">Test</ComplexWrapper>, { debug });

    expect(vNode).toMatchVDOM(
      <InlineComponent>
        <Component>
          <div>bar: Test</div>
        </Component>
      </InlineComponent>
    );
  });
```
This should fail.

Reply:
Thanks @ianlet 

@Varixo do you know how to fix this?

Reply:
not yet :D